components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -98,6 +99,13 @@ private void initializeDataPublisher() {
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
 
+        if (APIUtil.isAnalyticsEnabled()) {
+            WebSocketUtils.setApiPropertyToChannel(ctx, Constants.REQUEST_START_TIME_PROPERTY,
+                    System.currentTimeMillis());
```

Comment:
```suggestion
        if (APIUtil.isAnalyticsEnabled()) {
```

Reply:
Shall we remove adding this `REQUEST_START_TIME_PROPERTY` here. It's however overwritten later